modules/Sources/Features/TransactionList/TransactionListStore.swift

```diff
@@ -122,10 +122,7 @@ public struct TransactionListReducer: Reducer {
             
             let sortedTransactionList = transactionList
                 .sorted(by: { lhs, rhs in
-                    guard let lhsTimestamp = lhs.timestamp, let rhsTimestamp = rhs.timestamp else {
-                        return false
-                    }
-                    return lhsTimestamp > rhsTimestamp
+                    lhs.minedHeight ?? lhs.expiryHeight ?? 0 > rhs.minedHeight ?? rhs.expiryHeight ?? 0
```

Comment:
If I understand the Swift `??` chaining correctly, this is sorting by the first of these heights that is available:
1. Block height the transaction was mined at (if mined and known).
2. Expiry height for the transaction (even if set to zero).
   -  I don't actually recall whether zero can occur here or if the code that sets `.expiryHeight` elsewhere maps it to null.
4. Zero (i.e. the genesis block height, so it appears least recently in history).

AFAICT this means that mempool transactions (i.e. transactions you have just sent) will appear at the bottom of your transaction history instead of the top.

I think we instead want to sort by the first of these heights that is available:
1. Block height the transaction was mined at (if mined and known).
2. Expiry height for the transaction (if non-zero, which is always the case for transactions we create).
   - The "if non-zero" is important because an expiry height of 0 is not a real height (of the genesis block), but a sentinel for "no expiry height".
5. Mempool height (i.e. chain height + 1, so it appears most recently in history).
   - This is always available, so is the default fallback (and is what expired txs were defaulting to before).